src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java

```diff
@@ -169,27 +171,46 @@ public Consumer<Message<String>> consumeCancel() {
 
     protected abstract void saveResult(Network network, AbstractResultContext<R> resultContext, S result);
 
-    protected S run(Network network, AbstractResultContext<R> resultContext) throws Exception {
+    protected void logOnRun(R runContext) {
         LOGGER.info("Run {} computation ...", getComputationType());
+    }
+
+    /**
+     * if this computation needs extra run data, add them to the run context here
+     */
+    protected void enrichRunContext(R runContext) { }
 
-        R context = resultContext.runContext;
-        String provider = context.getProvider();
+    /**
+     * if this computation needs extra report operations, override this function and do them here
+     */
+    protected void reportSpecificOperations(R runContext, Reporter reporter) { }
+
+    protected S run(R runContext, UUID resultUuid) throws Exception {
+        logOnRun(runContext);
```

Comment:
Remove logOnRun and enrichRunContext and add a preRun method

Reply:
Done.